### PR TITLE
added SignOutButton Component with callbacks

### PR DESCRIPTION
**Pull request by omthakare16**

### CODE REVIEW

File: src/components/SignOutButton/SignOutButton.jsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import PropTypes from 'prop-types';
+import React from 'react';
+
+export const SignOutButton = ({callback, children}) => {
+  const auth = useEasyauth();
+  if (!children) {
+    return (
+      <a
+        href={
+          process.env.REACT_APP_EASYAUTH_APP_URL +
```
Comment (ghoshbishakh) [inline src/components/SignOutButton/SignOutButton.jsx:11]: take these from the initial configuration of EasyauthProvider somehow. Make useEasyauth access these somehow?

File: src/components/SignOutButton/SignOutButton.jsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+          '/logout?target=' +
+          btoa(callback || window.location.href)
+        }
+      >
+        <button
+          onClick={() => {
+            auth.removeUser();
```
Comment (ghoshbishakh) [inline src/components/SignOutButton/SignOutButton.jsx:18]: Try this: Configure removeUser such that after removing user it redirects to the logout URL.

Or.

Prepare a special page for logout, the page will only call removeUser.
In the logout button, just link to the logout url with redirect target to the special logout page.

Comment (omthakare16) [reply src/components/SignOutButton/SignOutButton.jsx:18]: @ghoshbishakh  , need  end_session_endpoint from .well-known/openid-configuration

File: src/components/SignOutButton/SignOutButton.jsx (hunk 1 above)
Comment (omthakare16) [reply src/components/SignOutButton/SignOutButton.jsx]: resolved in commit 2f7699f2d46f4a589f1b2438dfaf2866a8dd7c72.

File: src/components/SignOutButton/SignOutButton.jsx (hunk 2 above)
Comment (omthakare16) [reply src/components/SignOutButton/SignOutButton.jsx]: resolved in commit 2f7699f2d46f4a589f1b2438dfaf2866a8dd7c72.

**Review by ghoshbishakh — APPROVED** [review]

**Pull request merged**